### PR TITLE
fix: [EWT-80] Change `CustomerSegments` to array accordingly with PaymentV3 docs

### PR DESCRIPTION
**Pull request by Couper4**
This is a **breaking change** since the field is changed from `string` to `string[]`.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,10 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
+## [1.0.0] - 2023-01-23
+### Changed
+- `ProviderFilters.CustomerSegments` is now an array.
```
Comment (tl-mauro-franchi) [inline CHANGELOG.md:8]: I think the process is to add the changelog only once the new version is published, so you have to do it in a following PR. Also, when you do it, can you please add a link to the official changelong for context?

**Review by tl-mauro-franchi — COMMENTED** [review]
Aren't there any tests that need to be changed?

**Comment by Couper4** [comment]
> Aren't there any tests that need to be changed?

`ProviderFilter`'s fields aren't really tested, I've found only 3 reference of them in the tests ([1](https://github.com/TrueLayer/truelayer-dotnet/blob/d2e1b84f7151a733ea51f1c517018d2a8d962829/test/TrueLayer.Benchmarks/RequestSignatureBenchmarks.cs#L41), [2](https://github.com/TrueLayer/truelayer-dotnet/blob/d27d808fac3ed6b9a1dcfb23a06c6b7e134ec19a/test/TrueLayer.AcceptanceTests/MerchantAccountsTests.cs#L96), [3](https://github.com/TrueLayer/truelayer-dotnet/blob/a53ad1880b69d9d103613c5c62bcc447958329f8/test/TrueLayer.AcceptanceTests/PaymentTests.cs#L121)) and only `ProviderIds` is used.
Do you think it's worth adding some more tests?

File: CHANGELOG.md (hunk 1 above)
Comment (Couper4) [reply CHANGELOG.md:8]: Sure, I'll rollback these changes for now.

**Comment by tl-mauro-franchi** [comment]
> 
> `ProviderFilter`'s fields aren't really tested, I've found only 3 reference of them in the tests ([1](https://github.com/TrueLayer/truelayer-dotnet/blob/d2e1b84f7151a733ea51f1c517018d2a8d962829/test/TrueLayer.Benchmarks/RequestSignatureBenchmarks.cs#L41), [2](https://github.com/TrueLayer/truelayer-dotnet/blob/d27d808fac3ed6b9a1dcfb23a06c6b7e134ec19a/test/TrueLayer.AcceptanceTests/MerchantAccountsTests.cs#L96), [3](https://github.com/TrueLayer/truelayer-dotnet/blob/a53ad1880b69d9d103613c5c62bcc447958329f8/test/TrueLayer.AcceptanceTests/PaymentTests.cs#L121)) and only `ProviderIds` is used. Do you think it's worth adding some more tests?

Shame they are not really tested. If it is easy to add some tests then please do it especially given we have changed the type. I am just afraid that things might not work as expected. Of course, now it doesn't work anyway given the type is different from what the API is expecting it to be, but I think that if we test it now we might be able to catch any future issues.
What do you think?

**Comment by Couper4** [comment]
> > `ProviderFilter`'s fields aren't really tested, I've found only 3 reference of them in the tests ([1](https://github.com/TrueLayer/truelayer-dotnet/blob/d2e1b84f7151a733ea51f1c517018d2a8d962829/test/TrueLayer.Benchmarks/RequestSignatureBenchmarks.cs#L41), [2](https://github.com/TrueLayer/truelayer-dotnet/blob/d27d808fac3ed6b9a1dcfb23a06c6b7e134ec19a/test/TrueLayer.AcceptanceTests/MerchantAccountsTests.cs#L96), [3](https://github.com/TrueLayer/truelayer-dotnet/blob/a53ad1880b69d9d103613c5c62bcc447958329f8/test/TrueLayer.AcceptanceTests/PaymentTests.cs#L121)) and only `ProviderIds` is used. Do you think it's worth adding some more tests?
> 
> Shame they are not really tested. If it is easy to add some tests then please do it especially given we have changed the type. I am just afraid that things might not work as expected. Of course, now it doesn't work anyway given the type is different from what the API is expecting it to be, but I think that if we test it now we might be able to catch any future issues. What do you think?

Sure, I'll add some tests, it shouldn't be too much work.

**Comment by Couper4** [comment]
Thinking more about this: how should we test [ProviderFilter](https://github.com/TrueLayer/truelayer-dotnet/blob/main/src/TrueLayer/Payments/Model/ProviderFilter.cs)?
Actually it's a simple DTO class that gets serialized into JSON and sent along the HTTP request.
We would end up testing the serialization which doesn't make much sense to me.

Do you think there's a better way to do this?

**Comment by tl-mauro-franchi** [comment]
> Thinking more about this: how should we test [ProviderFilter](https://github.com/TrueLayer/truelayer-dotnet/blob/main/src/TrueLayer/Payments/Model/ProviderFilter.cs)? Actually it's a simple DTO class that gets serialized into JSON and sent along the HTTP request. We would end up testing the serialization which doesn't make much sense to me.
> 
> Do you think there's a better way to do this?
> 
> @tl-mauro-franchi

Yeah probably there is not much we can do about it. I also don't even know how that field is actually used and what changes in the response object based on the values you can pass to it (see [doc](https://docs.truelayer.com/reference/create-payment)) 🤔 🤷🏻 
Do you think it is worth asking the payments team about what we could test?

**Comment by Couper4** [comment]
> > Thinking more about this: how should we test [ProviderFilter](https://github.com/TrueLayer/truelayer-dotnet/blob/main/src/TrueLayer/Payments/Model/ProviderFilter.cs)? Actually it's a simple DTO class that gets serialized into JSON and sent along the HTTP request. We would end up testing the serialization which doesn't make much sense to me.
> > Do you think there's a better way to do this?
> > @tl-mauro-franchi
> 
> Yeah probably there is not much we can do about it. I also don't even know how that field is actually used and what changes in the response object based on the values you can pass to it (see [doc](https://docs.truelayer.com/reference/create-payment)) 🤔 🤷🏻 Do you think it is worth asking the payments team about what we could test?

I'm from the payment team! :grin:
`customer_segments` will just filter the providers. Testing that will imply mocking some data and then asserting on the response, which I personally think is an overkill, but let me know what do you think :wink:

**Comment by tl-mauro-franchi** [comment]
> I'm from the payment team! 😁 `customer_segments` will just filter the providers. Testing that will imply mocking some data and then asserting on the response, which I personally think is an overkill, but let me know what do you think 😉

Oh lol 😄 
Yeah I thought the same RE: mocking the data. In that case let's avoid it then 😉

**Review by tl-mauro-franchi — APPROVED** [review]

**Pull request merged**